### PR TITLE
 Introduce evmc_set_option_result enum

### LINKED ISSUE #102: set_option to have more response codes (success, fail, invalid option, invalid value)
Suggestion:
- 0: success
- 1: failure (general)
- 2: invalid option
- 3: invalid value for option

Perhaps it should return an enum?

### PR DESCRIPTION
**Pull request by axic**
Fixes #102.

### CODE REVIEW

File: include/evmc/evmc.h [hunk 1]
```diff
@@ -666,9 +675,11 @@ typedef void (*evmc_destroy_fn)(struct evmc_instance* evm);
  *  @param evm    The EVM instance to be configured.
  *  @param name   The option name. NULL-terminated string. Cannot be NULL.
  *  @param value  The new option value. NULL-terminated string. Cannot be NULL.
- *  @return       1 if the option set successfully, 0 otherwise.
+ *  @return       The outcome of the operation, @see ::evmc_set_option_result.
```
Comment (chfast) [inline include/evmc/evmc.h]: Remove `@see`.

File: test/vmtester/tests.cpp [hunk 2]
```diff
@@ -64,19 +64,19 @@ TEST_F(evmc_vm_test, set_option_unknown)
 {
     if (vm->set_option)
     {
-        int r = vm->set_option(vm, "unknown_option_csk9twq", "v");
-        EXPECT_EQ(r, 0);
+        enum evmc_set_option_result r = vm->set_option(vm, "unknown_option_csk9twq", "v");
+        EXPECT_NE(r, EVMC_SET_OPTION_SUCCESS);
```
Comment (chfast) [inline test/vmtester/tests.cpp]: The result should be "invalid name".

File: test/vmtester/tests.cpp [hunk 3]
```diff
@@ -64,19 +64,19 @@ TEST_F(evmc_vm_test, set_option_unknown)
...
         r = vm->set_option(vm, "unknown_option_csk9twq", "x");
-        EXPECT_EQ(r, 0);
+        EXPECT_NE(r, EVMC_SET_OPTION_SUCCESS);
```
Comment (chfast) [inline test/vmtester/tests.cpp]: The result should be "invalid name"

File: test/vmtester/tests.cpp [hunk 4]
```diff
@@ -64,19 +64,19 @@ TEST_F(evmc_vm_test, set_option_unknown)
...
 }
 
 TEST_F(evmc_vm_test, set_option_empty_value)
 {
     if (vm->set_option)
     {
-        int r = vm->set_option(vm, "unknown_option_csk9twq", nullptr);
-        EXPECT_EQ(r, 0);
+        enum evmc_set_option_result r = vm->set_option(vm, "unknown_option_csk9twq", nullptr);
+        EXPECT_NE(r, EVMC_SET_OPTION_SUCCESS);
```
Comment (chfast) [inline test/vmtester/tests.cpp]: "invalid name"

File: test/vmtester/tests.cpp (hunk 2 above)
Comment (axic) [reply test/vmtester/tests.cpp]: I had those here before, but figured it may better negating the expectation.

**Comment by axic** [comment]
@chfast updated.

**Comment by axic** [comment]
> You can add another unit test like this:

That will not work with Hera. I'd suggest, outside of this PR, that we define a standard option, which must be supported by all VMs (`debug` or `verbose`, with `on` and `off` and that can be tested)

**Pull request merged**